Adds a minor SM interaction with water vapor.

TemporalOroboros opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Makes the supermatter crystal split water vapor into hydrogen and oxygen above the power level it starts shocking things at. The efficiency increases with the crystals power.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
- Makes the spicy crystal a little more spicy.
- At high powers the supermatter starts launching lightning bolts. It just makes sense.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: The supermatter now splits water molecules at high power levels.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TemporalOroboros committed:
Electricity + 2H20 --> 2H2 + O2

Makes the supermatter electrolyze water vapor at high power levels.

Ghilker commented on code/modules/power/supermatter/supermatter.dm:
```diff
@@ -698,6 +700,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
+			var/h2o_moles = removed.gases[/datum/gas/water_vapor][MOLES]
+			var/electrolysis_moles = max(h2o_moles * (power - POWER_PENALTY_THRESHOLD)/(power + WATER_ELECTROLYSIS_MODIFIER), 0)
+			if(electrolysis_moles)
+				removed.gases[/datum/gas/water_vapor][MOLES] -= electrolysis_moles
```
use `removed.gases[/datum/gas/water_vapor][MOLES] -= min(h2o_moles, electrolysis_moles)` this way you'd be sure to not go under 0 (yes it should never go from how you set the electrolysis_moles math, but unless this is costly, better safe than sorry) or at least is what i would do..

TemporalOroboros replied:
I'll clamp it when I set up electrolysis moles.

Ghilker replied:
👍

Ghilker commented:
nice, little concerned about making the high level delamination faster since there will always be plasma fire, that make h2o that will produce h2 and o2 creating an even stronger fire that will repeat the cycle.

TemporalOroboros committed:
clamp

Changes a max operation to a clamp operation to prevent water vapor moles from being reduced below zero.

msgerbs commented:
I agree with Ghilker, but in theory this is an OK change. Would have to see what the end result is like.

Winterous commented:
There needs to be a net loss of oxygen in the cycle of burning, gotta be carefully balanced.

TemporalOroboros commented:
The supermatter already generates a surplus of oxygen. Even without this change the chamber reaches 70% oxygen on a heat delam.

wesoda25 commented:
Yeah the limiting factor from what I’ve seen is plasma, not oxygen, in shitty delams (usually born out of negligence).

LemonInTheDark commented:
How extensively has this been tested? I'm in support of more extreme delams, as it stands h2o has a higher effect on heat/plasma/o2 production then o2 does, so this might actually make things slower
